packages/grid/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx

```diff
@@ -78,11 +91,16 @@ function GridEditSingleSelectCell(props: GridRenderEditCellParams & Omit<SelectP
     });
   }
 
-  const handleChange = async (event: React.KeyboardEvent<HTMLInputElement>) => {
+  const handleChange: SelectProps['onChange'] = async (event) => {
     setOpen(false);
     const target = event.target as HTMLInputElement;
     // NativeSelect casts the value to a string.
     const formattedTargetValue = getValueFromValueOptions(target.value, valueOptionsFormatted);
+
+    if (onChange) {
+      await Promise.resolve(onChange(event, formattedTargetValue));
```

Comment:
Why is `Promise.resolve` needed here? This should work for both sync and async:

```suggestion
      await onChange(event, formattedTargetValue);
```

Reply:
Mainly to make explicit that `onChange` returns a promise. I removed it.